### PR TITLE
Adding Zephyr to list

### PR DESCRIPTION
**Pull request by benjaoming**
Let's get some cool projects on the list!

### CODE REVIEW

**Comment by benjaoming** [comment]
As always, the linting fails because the repo is <30 days old.

@ericholscher I wanted to hear your thoughts on this addition. I think it's a great example with lots of stuff to learn from. Also shows the scale of the platform... that's why I tried out a `#very-large-project` tag :)

File: README.md [hunk 1]
```diff
@@ -39,6 +40,10 @@ on [readthedocs.org](https://readthedocs.org/) and [readthedocs.com](https://rea
 - [sphinxcontrib-needs](https://sphinxcontrib-needs.readthedocs.io/en/latest/) - Documentation of `sphinxcontrib-needs`. **#sphinx** **#sphinx-themes** **#ui-material**.
 - [sphinx-immaterial](https://sphinx-immaterial.readthedocs.io/en/latest/) - Documentation of `sphinx-immaterial`, a Material theme for Sphinx, based on Material for MkDocs. **#sphinx** **#sphinx-themes** **#ui-material**.
 
+## API documentation
+
+- [Zephyr](https://docs.zephyrproject.org/latest/) - very large project: uses its own custom landing page, 404 pages and Doxygen+Breathe integration to document a whole operating system, written mostly in C. Contains a very large database tracking hardware support. **#sphinx** **#sphinx-themes** **#ui-material** **#doxygen** **#C** **#very-large-project**
```
Comment (ericholscher) [inline README.md:45]: I don't think we're hosting this project?

Comment (benjaoming) [reply README.md:45]: Oh gosh, I didn't even have the imagination that someone would go through so much effort....

https://github.com/zephyrproject-rtos/zephyr/blob/main/.github/workflows/doc-publish.yml
https://github.com/zephyrproject-rtos/zephyr/blob/main/.github/workflows/doc-build.yml

Comment (ericholscher) [reply README.md:45]: The world is a confusing place..

**Pull request closed without merging**